code/modules/mob/abstract/observer/observer.dm

```diff
@@ -159,6 +159,47 @@ Works together with spawning an observer, noted above.
 	if(medHUD)
 		process_medHUD(src)
 
+	teleport_if_needed()
+
+/mob/abstract/observer/proc/on_restricted_level(var/check)
+	if(!check)
+		check = z
+	//Check if they are a staff member
+	if(!check_rights(R_MOD|R_ADMIN|R_DEV, show_msg=FALSE, user=src))

```

Comment:
```suggestion
	if(check_rights(R_MOD|R_ADMIN|R_DEV, show_msg=FALSE, user=src))
```
psure this should be this way. IF HAVE RIGHTS -> return FALSE (NOT on a restricted level, do not teleport).